docs/internal/github_actions.rst

```diff
@@ -0,0 +1,195 @@
+job, which runs the appropriate tests for the associated OpenDDS
+configuration.
+
+Certain builds do not follow this 3 step model. Safety Profile builds are done
+in one step due to the cross-compile nature causing problems. Static and Release
+builds have a large footprint and therefore cannot fit the entire test suite onto
+a Github Actions runner.  As a result, they only build and run a subset of the tests
+in their final jobs, but then have multiple final jobs to increase test coverage. These
+jobs are prefixed by: *compiler_* which runs the ``tests/DCPS/Compiler tests``, *unit_*
+which runs the unit tests located in ``tests/DCPS/UnitTest`` and ``tests/unit-tests``, and
+*messenger_* which runs the tests in ``tests/DCPS/Messenger`` and ``tests/DCPS/C++11/Messenger``.
+
+In addition to these builds, there are some builds which will not run the test suite in
+an effort to shorten the runtime of the continuous integration.  An exception to this is
+that all builds which are not safety, and have ownership profile enabled, will run the
+``tests/cmake`` tests. Test runs which only contain CMake tests are prefixed by ``cmake_``.
+
+Test Results
+============
+
+The tests are run using `autobuild <https://github.com/DOCGroup/autobuild>`_ which creates a number of output files
+that are turned into a GitHub artifact. This artifact is processed by the
+"Check Test Results" workflow which modifies the files with detailed summaries of the test runs.
+After all of the Check Test Results jobs are complete, the test results will be posted in either
+the build_and_test or lint workflows. It is random which one of the workflows the results will appear
+in, so be sure to check both.
+
+Artifacts
+=========
+
+Artifacts from the continuous integration run can be downloaded by clicking details
+on one of the Build & Test runs. There is a dropdown on the bar next to "Re-run jobs"
+called "Artifacts" which lists each artifact that can be downloaded.
+
+Alternatively, clicking the "Summary" button at the top of the list of jobs will
+list all the available artifacts at the bottom of the page.
+
+Using Artifacts to Replicate Builds
+-----------------------------------
+
+You can download the ``ACE_TAO_`` and ``build_`` artifacts then use them for a local build,
+so long as your operating system is the same as the one on the runner.
+
+1. ``git clone`` the ACE_TAO branch which is targeted by the build. This is usually going to be
+``ace6tao2``.
+2. ``git clone --recursive`` the OpenDDS branch on which the CI was run.
+3. Merge OpenDDS master into your cloned branch.
+4. run ``tar xvfJ`` from inside the cloned ACE_TAO, targeting the *ACE_TAO_* .tar.xz file.
+5. run ``tar xvfJ`` from inside the cloned OpenDDS, targeting the *build_* .tar.xz file.
+6. Adjust the setenv.sh located inside OpenDDS to match the new locations for your ACE_TAO,
+and OpenDDS. The word "runner" should not appear within the setenv.sh once you are finished.
```

Comment:
[Not rendering as a list](https://opendds--2750.org.readthedocs.build/en/2750/internal/github_actions.html#using-artifacts-to-replicate-builds)

```suggestion
   ``ace6tao2``.
2. ``git clone --recursive`` the OpenDDS branch on which the CI was run.
3. Merge OpenDDS master into your cloned branch.
4. run ``tar xvfJ`` from inside the cloned ACE_TAO, targeting the *ACE_TAO_* .tar.xz file.
5. run ``tar xvfJ`` from inside the cloned OpenDDS, targeting the *build_* .tar.xz file.
6. Adjust the setenv.sh located inside OpenDDS to match the new locations for your ACE_TAO,
   and OpenDDS. The word "runner" should not appear within the setenv.sh once you are finished.
```